aten/src/ATen/native/Normalization.cpp

```diff
@@ -70,6 +70,74 @@ Tensor batch_norm(
             running_mean, running_var, training, momentum, eps);
 }
 
+Tensor layer_norm(const Tensor& input, IntList normalized_shape,
+    const Tensor& weight /* optional */, const Tensor& bias /* optional */,
+    double eps) {
+
+    int64_t normalized_ndim = normalized_shape.size();
+
+    if (normalized_ndim < 1) {
```

Comment:
other than this new check, this function is a line-by-line translation of the original python code (after removing the `running_*` related lines).

Reply:
The Python reference code is https://gist.github.com/SsnL/bce5d70236b6677cc9157fac8726eb05, which is copied from `nn/functional.py` and then with `running_*` related lines removed.